src/components/modal/modal.scss

```diff
@@ -184,13 +189,44 @@
  */
 .content {
   @apply relative box-border block h-full overflow-auto p-0;
-  max-block-size: 100%;
   background-color: var(--calcite-modal-content-background, theme("colors.background.foreground.1"));
-  padding-block: var(--calcite-modal-content-padding, var(--calcite-modal-padding-internal));
-  padding-inline: var(--calcite-modal-content-padding, var(--calcite-modal-padding-internal));
+  max-block-size: 100%;
+  padding: var(--calcite-modal-content-padding, var(--calcite-modal-padding-internal));
+}
+
+.content-top {
+  @apply bg-foreground-1
+  border-color-3
+  flex
+  min-w-0
+  max-w-full
+  rounded-t
```

Comment:
Is rounding needed for when there's no header nor footer?

Reply:
I made a correction here to have the top rounded on the `content-top` if there is no `header`, and the rounded bottom on the `content-bottom` if there is no `footer`.